docs-ref-conceptual/get-started-tutorial-3-use-variables.md

```diff
@@ -0,0 +1,305 @@
+
+Learning to store command output in a variable is beneficial when creating Azure resources which output secrets that should be protected. For example, when you create a service principal, reset a credential, or get an Azure key vault secret, the command output should be protected.
+
+Create a new Azure Key Vault and secret returning command output to a variable. Your Azure Key Vault name must be globally unique, so the `$RANDOM` identifier is used in this example. For more Azure Key Vault naming rules, see [Common error codes for Azure Key Vault](/azure/key-vault/general/common-error-codes).
+
+# [Bash](#tab/bash)
+
+```azurecli-interactive
+# Set your variables.
+let "randomIdentifier=$RANDOM*$RANDOM"
+rgName=<msdocs-tutorial-rg-00000000>
+kvName=msdocs-kv-$randomIdentifier
+location=eastus
+
+# Set your default output to none
+az config set core.output=none
+
+# Create a new Azure Key Vault returning the Key Vault ID
+myNewKeyVaultID=$(az keyvault create --name $kvName --resource-group $rgName --location $location --query id --output tsv)
+echo "My new Azure Kev Vault ID is $myNewKeyVaultID"
+
+# Create a new secret returning the secret ID
+kvSecretName=<myKVSecretName>
+kvSecretValue=<myKVSecretValue>
+myNewSecretID=$(az keyvault secret set --vault-name $kvName --name $kvSecretName --value $kvSecretValue --query id --output tsv)
+echo "My new secret ID is $myNewSecretID"
+
+# Reset your default output to json
+az config set core.output=json
+```
+
+# [PowerShell](#tab/powershell)
+
+```azurecli-interactive
+# Set your variables.
+$randomIdentifier=(New-Guid).ToString().Substring(0,8)
+$rgName="<msdocs-tutorial-rg-00000000>"
+$kvName="msdocs-kv-$randomIdentifier"
+$location="eastus"
+
+# Set your default output to none
+az config set core.output=none
+
+# Create a new Azure Key Vault returning the Key Vault ID
+$myNewKeyVaultID=$(az keyvault create --name $kvName --resource-group $rgName --location $location --query id --output tsv)
+echo "My new Azure Kev Vault ID is $myNewKeyVaultID"
+
+# Create a new secret returning the secret ID
+$kvSecretName="<myKVSecretName>"
+$kvSecretValue="<myKVSecretValue>"
+$myNewSecretID=$(az keyvault secret set --vault-name $kvName --name $kvSecretName --value $kvSecretValue --query id --output tsv)
+echo "My new secret ID is $myNewSecretID"
+
+# Reset your default output to json
+az config set core.output=json
+```
+
+---
+
+## Get the contents of a JSON file and store it in a variable
+
+This next section is a "stretch task" for an onboarding tutorial. However, to manage Azure resources in a development/stage/production scenario, you need to be able to read a configuration file.
+
+Are you ready to stretch your Azure CLI skills? Create a JSON file containing the following JSON, or your file contents of choice. Save the text file to you local drive. If you are working in Azure Cloud Shell, use the `upload/download files` icon in the menu bar to store the text file in your cloud storage drive.
+
+```text
```

Comment:
Is this not in a JSON format? I'm curious as to why this output is labeled "text"

Reply:
Good call.  b/c it was going to be a *.txt when I started. 👍